router.go

```diff
@@ -313,6 +314,92 @@ func (*App) copyRoute(route *Route) *Route {
 	}
 }
 
+func (app *App) normalizePath(path string) string {
+	if path == "" {
+		path = "/"
+	}
+	if path[0] != '/' {
+		path = "/" + path
+	}
+	if !app.config.CaseSensitive {
+		path = utils.ToLower(path)
+	}
+	if !app.config.StrictRouting && len(path) > 1 {
+		path = utils.TrimRight(path, '/')
+	}
+	return RemoveEscapeChar(path)
+}
+
+// RemoveRoute is used to remove a route from the stack by path.
+// This only needs to be called to remove a route, route registration prevents duplicate routes.
+// You should call RebuildTree after using this to ensure consistency of the tree.
+func (app *App) RemoveRoute(path string, methods ...string) {
+	// Normalize same as register uses
+	norm := app.normalizePath(path)
+
+	pathMatchFunc := func(r *Route) bool {
+		return r.path == norm // compare private normalized path
+	}
+	app.deleteRoute(methods, pathMatchFunc)
+}
+
+// RemoveRouteByName is used to remove a route from the stack by name.
+// This only needs to be called to remove a route, route registration prevents duplicate routes.
+// You should call RebuildTree after using this to ensure consistency of the tree.
+func (app *App) RemoveRouteByName(name string, methods ...string) {
+	matchFunc := func(r *Route) bool { return r.Name == name }
+	app.deleteRoute(methods, matchFunc)
+}
+
+// RemoveRouteFunc is used to remove a route from the stack by a custom match function.
+// This only needs to be called to remove a route, route registration prevents duplicate routes.
+// You should call RebuildTree after using this to ensure consistency of the tree.
+// Note: The route.Path is original path, not the normalized path.
+func (app *App) RemoveRouteFunc(matchFunc func(r *Route) bool, methods ...string) {
+	app.deleteRoute(methods, matchFunc)
+}
+
+func (app *App) deleteRoute(methods []string, matchFunc func(r *Route) bool) {
+	if len(methods) == 0 {
+		return // No methods provided, nothing to do
+	}
+
+	app.mutex.Lock()
+	defer app.mutex.Unlock()
+
+	removedUseRoutes := make(map[string]struct{})
+
+	for _, method := range methods {
+		// Uppercase HTTP methods
+		method = utils.ToUpper(method)
+
+		// Get unique HTTP method identifier
+		m := app.methodInt(method)
+		if m == -1 {
+			continue // Skip invalid HTTP methods
+		}
+
+		for i := len(app.stack[m]) - 1; i >= 0; i-- {
+			route := app.stack[m][i]
+			if !matchFunc(route) {
+				continue // Skip if route does not match
+			}
+
+			app.stack[m] = append(app.stack[m][:i], app.stack[m][i+1:]...)
+			app.routesRefreshed = true
+
+			// Decrement global handler count. In middleware routes, only decrement once
+			if _, ok := removedUseRoutes[route.path]; (route.use && slices.Equal(methods, app.config.RequestMethods) && !ok) || !route.use {
+				if route.use {
+					removedUseRoutes[route.path] = struct{}{}
+				}
+
+				atomic.AddUint32(&app.handlersCount, ^uint32(len(route.Handlers)-1)) //nolint:gosec // Not a concern
+			}
+		}
+	}
```

Comment:
we could also directly call the RebuildTree function , then the users will not forget it

Reply:
what do you think @gaby @sixcolors

Reply:
If you want to remove more than 1 route, then Rebuild will be called multiple times